assets/js/blocks/cart-checkout/mini-cart/component-frontend.tsx

```diff
@@ -13,25 +15,114 @@ import {
  * Internal dependencies
  */
 import CartLineItemsTable from '../cart/full-cart/cart-line-items-table';
+import './style.scss';
 
-const MiniCartContents = () => {
-	const { cartItems, cartIsLoading } = useStoreCart();
+interface MiniCartBlock {
+	isPlaceholderOpen?: boolean;
+}
 
-	if ( cartItems.length === 0 ) {
-		return <>{ __( 'Cart is empty', 'woo-gutenberg-products-block' ) }</>;
-	}
+const MiniCartBlock = ( { isPlaceholderOpen = false } ): JSX.Element => {
+	const { cartItems, cartItemsCount, cartIsLoading } = useStoreCart();
+	const [ isOpen, setIsOpen ] = useState< boolean >( isPlaceholderOpen );
+	// We already rendered the HTML drawer placeholder, so we want to skip the
+	// slide in animation.
+	const [ skipSlideIn, setSkipSlideIn ] = useState< boolean >(
+		isPlaceholderOpen
+	);
+
+	const contents =
+		cartItems.length === 0 ? (
+			<>{ __( 'Cart is empty', 'woo-gutenberg-products-block' ) }</>
+		) : (
+			<div className="is-mobile">
+				<CartLineItemsTable
+					lineItems={ cartItems }
+					isLoading={ cartIsLoading }
+				/>
+			</div>
+		);
 
 	return (
-		<CartLineItemsTable
-			lineItems={ cartItems }
-			isLoading={ cartIsLoading }
-		/>
+		<>
+			<button
+				className="wc-block-mini-cart__button"
+				onClick={ () => {
+					setIsOpen( true );
+					setSkipSlideIn( false );
+				} }
+			>
+				{ sprintf(
+					/* translators: %d is the count of items in the cart. */
+					_n(
+						'%d item',
+						'%d items',
+						cartItemsCount,
+						'woo-gutenberg-products-block'
+					),
+					cartItemsCount
+				) }
+			</button>
+			<Drawer
+				title={ sprintf(
+					/* translators: %d is the count of items in the cart. */
+					_n(
+						'Your cart (%d item)',
+						'Your cart (%d items)',
+						cartItemsCount,
+						'woo-gutenberg-products-block'
+					),
+					cartItemsCount
+				) }
+				isOpen={ isOpen }
+				onClose={ () => {
+					setIsOpen( false );
+				} }
+				slideIn={ ! skipSlideIn }
+			>
+				{ contents }
+			</Drawer>
+		</>
 	);
 };
 
-renderFrontend( {
-	selector: '.wc-block-mini-cart__contents',
-	Block: withStoreCartApiHydration(
-		withRestApiHydration( MiniCartContents )
-	),
-} );
+const renderMiniCartFrontend = () => {
+	// Check if button is focused. In that case, we want to refocus it after we
+	// replace it with the React equivalent.
+	let focusedMiniCartBlock: HTMLElement | null = null;
+	/* eslint-disable @wordpress/no-global-active-element */
+	if (
+		document.activeElement &&
+		document.activeElement.classList.contains(
+			'wc-block-mini-cart__button'
+		) &&
+		document.activeElement.parentNode instanceof HTMLElement
+	) {
+		focusedMiniCartBlock = document.activeElement.parentNode;
+	}
+	/* eslint-enable @wordpress/no-global-active-element */
+
+	renderFrontend( {
+		selector: '.wc-block-mini-cart',
+		Block: withStoreCartApiHydration(
+			withRestApiHydration( MiniCartBlock )
```

Comment:
Yes, I noticed that as well. I opened an issue to track this, because I think the issue might be in the GB component we use: https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/4618.